crates/bevy_asset/src/server/info.rs

```diff
@@ -404,7 +405,10 @@ impl AssetInfos {
                         // If dependency is loaded, reduce our count by one
                         loading_rec_deps.remove(dep_id);
                     }
-                    RecursiveDependencyLoadState::Failed => {
+                    RecursiveDependencyLoadState::Failed(ref error) => {
+                        if let None = first_rec_error {
+                            first_rec_error = Some(error.clone())
+                        }
```

Comment:
note: the `retain` loop is not ordered, so this might not actually be the first recursive error chronologically. I don't think this really matters though

Reply:
Yeah it's async: ordering is very much an illusion.